data/core/common.lua

```diff
@@ -273,4 +273,26 @@ function common.relative_path(ref_dir, dir)
 end
 
 
+function common.mkdirp(path)
+  local stat = system.get_file_info(path)
+  if stat and stat.type then
```

Comment:
isn't checking for `stat` and `stat.type` redundant?

Reply:
I just copied what you wrote in you version of mkdirp without thinking too much about that. I thought you had a good reason for doing this.